.github/workflows/cdelivery-s3-apps.yml

```diff
@@ -0,0 +1,82 @@
+name: CD reusable workflow for S3 apps
+
+# Control when the action will run.
+on:
+  # Allow the workflow to be reusable
+  workflow_call:
+    inputs:
+      build-folder:
+        required: false
+        default: 'build'
+        type: string
+      version:
+        required: false
+        default: 'latest'
+        type: string
+    secrets:
+      aws-access-key-id:
+        required: true
+      aws-secret-access-key:
+        required: true
+      aws-region:
+        required: true
+      aws-s3-bucket-name:
+        required: true
+      app-id:
+        required: true
+      graasp-domain-prod:
+        required: true
+      cloudfront-distribution:
+        required: true
+
+env:
+  BUILD_FOLDER: '${{ inputs.build-folder }}'
+  VERSION: '${{ inputs.version }}'
+
+# This workflow contains one job called: build-deploy
+jobs:
+  build-deploy:
+    name: Build, sync with S3 and deploy to prod
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v2
+
+      - name: Setup node
+        id: setup-node
+        uses: actions/setup-node@v2
+        with:
+          node-version: '16'
+          check-latest: true
+
+      - name: Yarn build prod
+        id: build-image
+        env: 
+          REACT_APP_GRAASP_DOMAIN: ${{ secrets.graasp-domain-prod }}
+          REACT_APP_GRAASP_APP_ID: ${{ secrets.app-id }}
+          REACT_APP_MOCK_API: false
+        run: |
+          yarn install
+          yarn run build
+
+      - name: Configure AWS Credentials
+        id: configure-aws
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-access-key-id: ${{ secrets.aws-access-key-id }}
+          aws-secret-access-key: ${{ secrets.aws-secret-access-key }}
+          aws-region: ${{ secrets.aws-region }}
+
+      - name: Sync files with AWS S3 Bucket
+        id: sync-s3
+        env:
+          APP_DIR: '${{ secrets.aws-s3-bucket-name }}/${{ secrets.app-id }}/${{env.VERSION}}/'
```

Comment:
How is the version decided? Is it latest by default? I don't know what would be the strategy for apps, push to latest and some tag? @juancarlosfarah

Reply:
the version is set to latest by default, but can be specified from the caller workflow of each app and set to a different value.